File: solgen.portal/ClientApp/src/app/views/shared/notes/notes.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../../environments/environment';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class NotesService {
  public notesUri = `${environment.WebApiBaseUrl}Notes`;
  pagedData: any;
  pagedDataNewCardView: any;
  constructor(private http: HttpClient) { }


  getNotesList(notesComment: string, sortColumn: string, sortDir, page: number, pageSize: number, userId: string): Observable<any> {
    return this.http.get(`${this.notesUri}/GetList?notesComment=${notesComment}&sortColumn=${sortColumn}&sortDir=${sortDir}&page=${page}&pageSize=${pageSize}&userId=${userId}`)
      .pipe(
        map((response: any) => {
          this.pagedData = response;
          return true;
        })
      );
  }

  saveNotes(note: Notes, contactId: any) {
    if (contactId != null) {
      note.notesAddedForId = contactId;
    }
    return this.http.post(this.notesUri, note);
  }
  saveNewNotes(note: FormData) {
    return this.http.post(this.notesUri + `/NewNoteSave`,  note);
  }
  getNoteslist(leadid: any, searchText: any, submodulename: any, objectname: any, sortColumn: string, sortDir: string, page: number, pageSize: number, userId: string, view: boolean=false): Observable<any> {
    return this.http.get(`${this.notesUri}/getNoteslist?leadid=${leadid}&searchText=${searchText}&submodulename=${submodulename}&objectname=${objectname}&SortColumn=${sortColumn}&SortDir=${sortDir}&PageNo=${page}&PageSize=${pageSize}&userId=${userId}&view=${view}`)
      .pipe(
        map((response: any) => {
          this.pagedData = response;
          return true;
        })
      );
  }

  getNoteslistCardView(leadid: any, submodulename: any, objectname: any, sortColumn: string, sortDir: string, page: number, pageSize: number, userId: string): Observable<any> {
    return this.http.get(`${this.notesUri}/getNoteslistForCardView?leadid=${leadid}&submodulename=${submodulename}&objectname=${objectname}&SortColumn=${sortColumn}&SortDir=${sortDir}&PageNo=${page}&PageSize=${pageSize}&userId=${userId}`)
      .pipe(
        map((response: any) => {
          this.pagedDataNewCardView = response;
          return true;
        })
      );
  }

  GetImageList(noteId: any, modulename: any, objectRefId: any) {
    return this.http.get(this.notesUri + `/GetImageList?noteId=${noteId}&modulename=${modulename}&objectRefId=${objectRefId}`)
  }
  GetNoteThreadList(noteId: any,  objectRefId: any) {
    return this.http.get(this.notesUri + `/getNoteslistForNotesThreadsByNoteId?leadid=${objectRefId}&notesId=${noteId}`)
  }
  GetNoteCurentDataList(noteId: any, objectRefId: any) {
    return this.http.get(this.notesUri + `/GetNoteCurentDataList?leadid=${objectRefId}&notesId=${noteId}`)
  }
  ChangeStatusForAddmin(noteId: any, objectRefId: any, status:any) {
    return this.http.get(this.notesUri + `/GetChangeStatusForAddmin?objectRefId=${objectRefId}&notesId=${noteId}&status=${status}`)
  }
  setNotesToRead(noteid: any) {
    return this.http.get(`${this.notesUri}/SetNotesToRead?noteid=${noteid}`)
  }
  delImage(Id:any,noteId: any, modulename: any, objectRefId: any) {
    return this.http.get(this.notesUri + `/deleteMulipleUploadingImageImage?noteId=${noteId}&modulename=${modulename}&objectRefId=${objectRefId}&Id=${Id}`)
  }
  GetWorkOrderTypeBasedonWorkorder(accountId:any, objectRefId:any,submoduleName:any) {
    return this.http.get(this.notesUri + `/GetWorkOrderTypeBasedonWorkorder?accountId=${accountId}&objectRefId=${objectRefId}&submoduleName=${submoduleName}`);
  }
  DeleteNote(Id: any ) {
    return this.http.get(this.notesUri + `/DeleteNote?Id=${Id}`)
      
  }
  PinNotes(Id: any) {
    return this.http.get(this.notesUri + `/PinNote?Id=${Id}`)

  }
  private initializeNotes(): Notes {
    // Return an initialized object
    return {
      notesComments: null,
      notesHistoryId: null,
      notesAddedById: null,
      notesCreatedOn: null,
      notesIsDeleted: false,
      notesAddedForId: null,
      name: null,
      userType: null
    };
  }
}

export class Notes {
  notesComments: string;
  notesHistoryId: string;
  notesAddedById: string;
  notesCreatedOn: string;
  notesIsDeleted: boolean;
  notesAddedForId: string;
  name: string;
  userType: string;

  constructor() {
    this.notesComments = '';
    this.notesHistoryId = '';
    this.notesAddedById = '';
    this.notesCreatedOn = '';
    this.notesIsDeleted = false
    this.notesAddedForId = '';
    this.name = '';
    this.userType = '';

  }
 
}
export class newNotesModel {
  noteid: number;
  objectrefid: number;
  title: string;
  status: string;
  assignTo: string;
  notesComments: string;
  moduleName: string;
  RouteUrl:string;
  submoduleName: string;
  notifyTo: string;
  constructor() {
    this.title = '';
    this.status = '';
    this.assignTo = '';
    this.notesComments = '';
    this.moduleName = '';
    this.submoduleName = '';
    this.noteid = 0;
    this.notifyTo = '';
    

  }
}

File: solgen.Service/Notes/NotesService.cs
using System;
using System.Collections.Generic;
using System.Text;
using Solgen.Core;
using Solgen.Repository;

namespace Solgen.Service
{
    public class NotesService: INotesService  
    {
        private readonly INotesRepository repository;

        public NotesService(INotesRepository repository)
        {
            this.repository = repository;
        }
        /*! 
 * \brief   Add notes
 * \details  Get Module List
 * \author  Deepak jha 
 * \date    16 Sept 2019
 * \version 1.0    
 */
        public Guid Save(NotesHistory entity)
        {
            try
            {
                return repository.Save(entity);
            }
            catch (Exception ex)
            {

                throw new Exception(ex.Message);
            }
        }
        /*! 
 * \brief   GetList
 * \details  Get notes List
 * \author  Deepak jha 
 * \date    16 Sept 2019
 * \version 1.0    
 */
        public PagedData GetList(string notesComment, string sortColumn, string sortDir, int pageIndex, int pageSize, Guid? userId)
        {
            try
            {
                return repository.GetList(notesComment, sortColumn, sortDir, pageIndex, pageSize, userId);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
        }

        public int NewNoteSave(Notes note)
        {
            try
            {
                return repository.NewNoteSave(note);
            }
            catch (Exception ex)
            {

                throw new Exception(ex.Message);
            }
        }
        public PagedData getNoteslist(long leadid, string searchText, string submodulename, string objectname, string SortColumn, string SortDir, int PageNo, int PageSize, Guid? userId, long companyID, Boolean view)
        {
            return repository.getNoteslist(leadid, searchText, submodulename, objectname, SortColumn, SortDir, PageNo, PageSize, userId, companyID, view);
        }
        public PagedData getNoteslistForCardView(long leadid, string submodulename, string objectname, string SortColumn, string SortDir, int PageNo, int PageSize, Guid? userId, long companyID)
        {
            return repository.getNoteslistForCardView(leadid,  submodulename, objectname, SortColumn, SortDir, PageNo, PageSize, userId, companyID);
        }
        public int DeleteNote(string id, Guid? userId)
        {
            try
            {
                return repository.DeleteNote(id,userId);
            }
            catch (Exception ex)
            {

                throw new Exception(ex.Message);
            }
        }
        public string  SetNotesToRead(string noteid, string companyId)
        {
            try
            {
                return repository.SetNotesToRead(noteid, companyId);
            }
            catch (Exception ex)
            {

                throw new Exception(ex.Message);
            }
        }

        public int  SaveFileUploadingDetails(List<UploadFileModelList> list)
        {
            try
            {
                return repository.SaveFileUploadingDetails(list);
            }
            catch (Exception ex)
            {

                throw new Exception(ex.Message);
            }
        }
        public List<UploadFileModelList> GetNoteImageList(string noteId, string modulename, string objectRefId, string companyID, string userId)
        {
            try
            {
                return repository.GetNoteImageList(noteId, modulename, objectRefId, companyID,  userId);
            }
            catch (Exception ex)
            {

                throw new Exception(ex.Message);
            }
        }
        public int deleteMulipleUploadingImageImage(string Id, string noteId, string modulename, string objectRefId, string companyID, string userId)
        {
            try
            {
                return repository.deleteMulipleUploadingImageImage(Id,noteId, modulename, objectRefId, companyID, userId);
            }
            catch (Exception ex)
            {

                throw new Exception(ex.Message);
            }
        }

        public IEnumerable<dynamic> GetWorkOrderTypeBasedonWorkorder(string accountId, string objectRefId, string userId, string companyId, string submoduleName)
        {
            return repository.GetWorkOrderTypeBasedonWorkorder( accountId,  objectRefId,  userId,  companyId,submoduleName);
        }
        public string  getNoteslistForCardViewgetNoteslistForNotesThreadsByNoteId(long leadid, string NotesId, long companyID, string submodulename, string objectname, string SortColumn, string SortDir, int PageNo, int PageSize, Guid? userId)
        {
            return repository.getNoteslistForCardViewgetNoteslistForNotesThreadsByNoteId(leadid,  NotesId,  companyID, submodulename,  objectname,  SortColumn,  SortDir,  PageNo,  PageSize,  userId);
        }
        public string GetNoteCurentDataList(string leadid, string NotesId, string companyID, Guid? userId)
        {
            return repository.GetNoteCurentDataList(leadid,  NotesId,  companyID,  userId);
        }
        public string GetChangeStatusForAddmin(string objectRefId, string userId, string companyId, string notesId, string status)
        {
            return repository.GetChangeStatusForAddmin(objectRefId, userId, companyId, notesId,status);
        }
        public int PinNote(string id, string userId, long companyID)
        {
            return repository.PinNote( id,  userId,  companyID);
        }
    }
}
